### ISSUE TITLE
Defining a polygon restriction in load_collection and mask_polygon gives some strange extra data

### ISSUE THREAD

**Issue by danielFlemstrom** [Open-EO/openeo-web-editor#252]
When defining a geojson polygon in the web editor, like this:

![image](https://user-images.githubusercontent.com/12410550/168003839-bd13015b-3c88-4338-8d48-0de4e616005d.png)

we got some extra data at the end that looks a bit suspicious (the "correct" polygon removed)
![image](https://user-images.githubusercontent.com/12410550/168005549-aec9ec16-3f73-41fb-bd26-0326e5371074.png)

As for now, the generated code just omits the extent ( https://github.com/Open-EO/openeo-odc/issues/3) but it would be nice to know if the web-editor is producing the correct graph or not since the polygon handling is important to us.  

Depending on how you have decided, maybe remove the option from load collection and let mask_polygon do the cutting altogether. Unfortunately, when i tested to use the mask_polygon process,  that process got the same strange structure.

**Comment by m-mohr** [comment]
Hmm, I can't reproduce this right now. The second Feature indeed looks weird, because it is simply a point basically. But the OpenLayers GeoJSON Editor is sometimes a bit weird, indeed, so I'm not surprised.

> As for now, the generated code just omits the extent ( https://github.com/Open-EO/openeo-odc/issues/3) but it would be nice to know if the web-editor is producing the correct graph or not since the polygon handling is important to us.

I'd assume the first one is correct and the second one should not be there. Can you reproduce this reliably?

> Depending on how you have decided, maybe remove the option from load collection and let mask_polygon do the cutting altogether. Unfortunately, when i tested to use the mask_polygon process, that process got the same strange structure.

The options will not be removed from load_collection. Ultimately, the code that produces the GeoJSON is anyway the same so mask_polygon should struggle from the same behavior in the Web Editor. I'm not sure about the ODC issues, I'm not involved there.

**Comment by danielFlemstrom** [comment]
yes, I tried again and get the same behavior, the following geojson:
![image](https://user-images.githubusercontent.com/12410550/171107213-af625d55-f9cc-4ce7-8f4d-459bd5ac08ee.png)

Yields the following code:
{
  "type": "FeatureCollection",
  "features": [
    {
      "type": "Feature",
      "geometry": {
        "type": "Polygon",
        "coordinates": [
          [
            [
              16.516218310221195,
              59.584669076046936
            ],
            [
              16.560443627213292,
              59.60625222491251
            ],
            [
              16.625202127094575,
              59.59026603768825
            ],
            [
              16.6362584563426,
              59.55346894938421
            ],
            [
              16.615725273453414,
              59.526245234032245
            ],
            [
              16.560443627213292,
              59.52384208542571
            ],
            [
              16.516218310221195,
              59.584669076046936
            ]
          ]
        ]
      },
      "properties": null
    },
    {
      "type": "Feature",
      "geometry": {
        "type": "Polygon",
        "coordinates": [
          [
            [
              16.516218310221195,
              59.584669076046936
            ],
            [
              16.516218310221195,
              59.584669076046936
            ],
            [
              16.516218310221195,
              59.584669076046936
            ],
            [
              16.516218310221195,
              59.584669076046936
            ]
          ]
        ]
      },
      "properties": null
    }
  ]
}

**Comment by danielFlemstrom** [comment]
The problem seems to be solved now!

**Comment by m-mohr** [comment]
Thanks for following up, seemed to be an upstream issue then.

**Comment by danielFlemstrom** [comment]
Sorry just discovered that using the :latest docker container instead of the web deployed openeo-web-editor, the problem is still there. 
If this is due to an outdated :latest, you can close this issue again. 
I cannot test the online web-editor right now, so i cannot confirm that the problem exists there as well.

**Comment by m-mohr** [comment]
Not sure, it looks up to date, but I'm neither maintaining nor using the Docker image. 

Also, this looks like an upstream issue in ol or ol-ext. So not sure I can do a lot about it.

**Comment by m-mohr** [comment]
I was able to reproduce it. If you try to finish the bbox and try that via a lot of clicking on the same point, such a polygon that looks like a point is created.

The EditBar is not super easy to use honestly, but it is primarily a "user error". I'm not sure whether there's anything we can do to improve this.
